src/filters.rs

```diff
@@ -60,6 +60,46 @@ pub fn upcase(input: &Value, _args: &[Value]) -> FilterResult {
     }
 }
 
+
+pub fn downcase(input: &Value, _args: &[Value]) -> FilterResult {
+    match *input {
+        Str(ref s) => Ok(Str(s.to_lowercase())),
+        _ => Err(InvalidType("String expected".to_owned())),
+    }
+}
+
+
+pub fn capitalize(input: &Value, _args: &[Value]) -> FilterResult {
+    match *input {
+        Str(ref s) => Ok(Str(s.char_indices().fold(String::new(), |word, (_, chr)| {
+            let next_char = match word.chars().last() {
+                Some(last) =>
+                    if last.is_whitespace() {
+                        chr.to_uppercase().next().unwrap()
+                    } else {
+                        chr
+                    },
+                _ => chr.to_uppercase().next().unwrap(),
+            }.to_string();
+            word + &next_char
```

Comment:
I would've done something like

```
        Str(ref s) => Ok(Str(s.split(' ').map(|word| {
            let (h, t) = word.split_at(1);
            h.to_uppercase() + t
        }).collect::<Vec<String>>().join(" "))),
```
but it seems you're trying to preserve whitespace, which is fine by me. Can I just ask you to add some comments explaining your code, because I really can't wrap my head around it. Also, are the `unwrap` calls guaranteed to be safe?